src/main/java/io/github/opencubicchunks/cubicchunks/world/level/cube/ProtoCube.java

```diff
@@ -1,10 +1,191 @@
+    @Override public native Holder<Biome> getNoiseBiome(int pX, int pY, int pZ);
+
+    @TransformFrom(copyFrom = @CopyFrom(clazz = ProtoChunk.class), value = "packOffsetCoordinates(Lnet/minecraft/core/BlockPos;)S")
+    public native static short packOffsetCoordinates(BlockPos pPos);
+
+    @TransformFrom(copyFrom = @CopyFrom(clazz = ProtoChunk.class), value = "unpackOffsetCoordinates(SILnet/minecraft/world/level/ChunkPos;)Lnet/minecraft/core/BlockPos;")
+    public native static BlockPos unpackOffsetCoordinates(short pPackedPos, int pYOffset, ChunkPos pChunkPos);
+
+    // dasm + mixin
+    @TransformFrom(copyFrom = @CopyFrom(clazz = ProtoChunk.class), value = "markPosForPostprocessing(Lnet/minecraft/core/BlockPos;)V")
+    @Override public native void markPosForPostprocessing(BlockPos pPos);
+
+    @TransformFrom(copyFrom = @CopyFrom(clazz = ProtoChunk.class), value = "addPackedPostProcess(SI)V")
+    @Override public native void addPackedPostProcess(short pPackedPosition, int pIndex);
+
+    @TransformFrom(copyFrom = @CopyFrom(clazz = ProtoChunk.class), value = "getBlockEntityNbts()Ljava/util/Map;")
+    @Override public native Map<BlockPos, CompoundTag> getBlockEntityNbts();
+
+    @TransformFrom(copyFrom = @CopyFrom(clazz = ProtoChunk.class), value = "getBlockEntityNbtForSaving(Lnet/minecraft/core/BlockPos;)Lnet/minecraft/nbt/CompoundTag;")
+    @Override @Nullable public native CompoundTag getBlockEntityNbtForSaving(BlockPos pPos);
+
+    @TransformFrom(copyFrom = @CopyFrom(clazz = ProtoChunk.class), value = "removeBlockEntity(Lnet/minecraft/core/BlockPos;)V")
+    @Override public native void removeBlockEntity(BlockPos pPos);
+
+    @TransformFrom(copyFrom = @CopyFrom(clazz = ProtoChunk.class), value = "getCarvingMask(Lnet/minecraft/world/level/levelgen/GenerationStep$Carving;)Lnet/minecraft/world/level/chunk/CarvingMask;")
+    @Override @Nullable public native CarvingMask getCarvingMask(GenerationStep.Carving pStep);
+
+    @TransformFrom(copyFrom = @CopyFrom(clazz = ProtoChunk.class), value = "getOrCreateCarvingMask(Lnet/minecraft/world/level/levelgen/GenerationStep$Carving;)Lnet/minecraft/world/level/chunk/CarvingMask;")
+    @Override public native CarvingMask getOrCreateCarvingMask(GenerationStep.Carving pStep);
+
+    @TransformFrom(copyFrom = @CopyFrom(clazz = ProtoChunk.class), value = "setCarvingMask(Lnet/minecraft/world/level/levelgen/GenerationStep$Carving;Lnet/minecraft/world/level/chunk/CarvingMask;)V")
+    @Override public native void setCarvingMask(GenerationStep.Carving pStep, CarvingMask pCarvingMask);
+
+    @TransformFrom(copyFrom = @CopyFrom(clazz = ProtoChunk.class), value = "setLightEngine(Lnet/minecraft/world/level/lighting/LevelLightEngine;)V")
+    @Override public native void setLightEngine(LevelLightEngine pLightEngine);
+
+    @Override public void setBelowZeroRetrogen(@Nullable BelowZeroRetrogen pBelowZeroRetrogen) {
+        // Unused
+    }
```

Comment:
Thoughts on making this dasm?

Reply:
The original method body actually sets it; I'm replacing it with an empty method body. Will add a comment to make that clearer